client/app/components/app-view/index.js

```diff
@@ -1,8 +1,14 @@
 import debug from 'debug';
+import { matchesUA, resolveUserAgent } from 'browserslist-useragent';
 import PromiseRejectionError from '@/lib/promise-rejection-error';
 import { ErrorHandler } from './error-handler';
 import template from './template.html';
 
+if (!matchesUA(navigator.userAgent, { allowHigherVersions: true })) {
+  // TODO: create "browser outdated" page and redirect to it
+  console.log('Your browser is outdated', resolveUserAgent(navigator.userAgent));
```

Comment:
Is this the place (and method) to put the redirect?

Reply:
Yes, it's the right place. And probably you don't need a page redirect: create a component with message or whatever, and conditionally show it instead of `<div ng-view>`

Reply:
But then we're still running potentially unsupported js for that page.
If we early redirect to a static html page, we don't have to worry about it.

Reply:
If you detect UA on client-side - you always have a chance to run unsupported code. If you'll change `app-view/template.html` to something like this - no change it will render header/footer/page:
```html
<div ng-if="$ctrl.isBrowserSupported">
  <!-- original template here -->
</div>
<div ng-if="!$ctrl.isBrowserSupported">
  Error! Error!!!! Unsupported browser!!!!111
</div>
```

Reply:
It's important that `app-view` component itself shouldn't use any unsupported features, as well as all code it uses via imports.

Reply:
Sounds easy and simple.
I'll do that and also add a cypress test to make sure it doesn't break.

Reply:
@kravets-levko does this make sense? (I'd rather it have it's own template)
```js
export default function init(ngModule) {
  ngModule.factory(
    '$exceptionHandler',
    () => function exceptionHandler(exception) {
      handler.process(exception);
    },
  );

  const isBrowserSupported = someCheck(...);
  ngModule.component('appView', {
    template: isBrowserSupported ? template : notSupportedTemplate,
    controller: isBrowserSupported ? AppViewComponent : NotSupportedCtrl,
  });
}
```

Reply:
@ranbena Yes, it's probably even better :+1: